src/gui/addnewtorrentdialog.cpp

```diff
@@ -399,23 +401,22 @@ void AddNewTorrentDialog::showAdvancedSettings(bool show)
 
 void AddNewTorrentDialog::saveSavePathHistory() const
 {
-    QDir selectedSavePath(m_ui->savePath->selectedPath());
     // Get current history
     QStringList history = settings()->loadValue(KEY_SAVEPATHHISTORY).toStringList();
-    if (history.size() > savePathHistoryLength())
-        history = history.mid(0, savePathHistoryLength());
-    QList<QDir> historyDirs;
-    foreach (const QString dir, history)
-        historyDirs << QDir(dir);
-    if (!historyDirs.contains(selectedSavePath)) {
-        // Add save path to history
+    QVector<QDir> historyDirs;
+    for (const QString path : qAsConst(history))
```

Comment:
just asking, is it your intention for not using QString reference?

Reply:
This was in the previous code. I just wasn't paying attention. I'll fix it.

Reply:
Done.